### PR TITLE
Jonmv/reindexing metrics

### PR DESCRIPTION
**Pull request by jonmv**
@bjorncs please review.

### CODE REVIEW

File: clustercontroller-reindexer/src/main/java/ai/vespa/reindexing/ReindexingMetrics.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    private final String cluster;
+
+    ReindexingMetrics(Metric metric, String cluster) {
+        this.metric = metric;
+        this.cluster = cluster;
+    }
+
+    void dump(Reindexing reindexing) {
+        reindexing.status().forEach((type, status) -> {
+            metric.set("reindexing.percent.done",
```
Comment (bjorncs) [inline clustercontroller-reindexer/src/main/java/ai/vespa/reindexing/ReindexingMetrics.java:29]: Consider having "percent" removed from metric name, and instead represent it as a value between 0 and 1.

**Review by bjorncs — APPROVED** [review]

File: clustercontroller-reindexer/src/main/java/ai/vespa/reindexing/ReindexingMetrics.java (hunk 1 above)
Comment (geirst) [reply clustercontroller-reindexer/src/main/java/ai/vespa/reindexing/ReindexingMetrics.java:29]: Also, we should discuss the metric names in an architect meeting at some time.

Comment (jonmv) [reply clustercontroller-reindexer/src/main/java/ai/vespa/reindexing/ReindexingMetrics.java:29]: Yes, that’s the intention :) Feel free to reach out early with suggestions.

**Comment by jonmv** [comment]
Actually metrics and names will wait until discussed properly.

File: clustercontroller-reindexer/src/main/java/ai/vespa/reindexing/Reindexer.java [hunk 2]
```diff
@@ -110,20 +107,20 @@ public void reindex() throws ReindexingLockException {
...
-                                                        .running()
-                                                        .successful(clock.instant()));
-        if (ready.get(type).isAfter(status.startedAt()))
-            status = Status.ready(clock.instant()); // Need to restart, as a newer reindexing is required.
-
-        database.writeReindexing(reindexing = reindexing.with(type, status));
-        metrics.dump(reindexing);
-
-        switch (status.state()) {
+        AtomicReference<Reindexing> reindexing = new AtomicReference<>(database.readReindexing());
```
Comment (bjorncs) [inline clustercontroller-reindexer/src/main/java/ai/vespa/reindexing/Reindexer.java:110]: You should probably use a shared monitor/lock for both `reindexing` and `status` (unless they are used completely  independently of each other).

**Review by bjorncs — APPROVED** [review]

File: clustercontroller-reindexer/src/main/java/ai/vespa/reindexing/Reindexer.java (hunk 2 above)
Comment (jonmv) [reply clustercontroller-reindexer/src/main/java/ai/vespa/reindexing/Reindexer.java]: They are used independently enough imo.

**Comment by jonmv** [comment]
Second last commit makes sure the state of all pending reindexing are set to pending when a new directive is received, rather than exactly when reindexing of that type begins.

**Review by bjorncs — APPROVED** [review]

**Pull request merged**